### ISSUE TITLE
Passing variables to partial does not work when model is specified in 4.0.0 and later

### ISSUE THREAD

**Issue by forbjok** [sunng87/handlebars-rust#519]
The following worked fine in 3.x:

```rust
use handlebars::Handlebars;
use serde_derive::Serialize;

#[derive(Serialize)]
struct Data {
    a: String,
}

fn main() {
    let mut reg = Handlebars::new();

    reg.register_template_string("sub", "SUB {{a}} // {{something}} BUS")
        .unwrap();

    let data = Data { a: "TEST".into() };

    // something gets passed through correctly
    dbg!(reg
        .render_template("{{> sub something=\"SOMETHING\"}}", &data)
        .unwrap());

    // something does not get passed through
    dbg!(reg
        .render_template("{{> sub this something=\"SOMETHING\"}}", &data)
        .unwrap());
}
```

Expected result, as seen in 3.x:
```
[src\main.rs:18] reg.render_template("{{> sub something=\"SOMETHING\"}}", &data).unwrap() = "SUB TEST // SOMETHING BUS"
[src\main.rs:23] reg.render_template("{{> sub this something=\"SOMETHING\"}}", &data).unwrap() = "SUB TEST // SOMETHING BUS"
```

However, starting with 4.0.0, the variable "something" becomes blank if a model is specified.

```
[src\main.rs:18] reg.render_template("{{> sub something=\"SOMETHING\"}}", &data).unwrap() = "SUB TEST // SOMETHING BUS"
[src\main.rs:23] reg.render_template("{{> sub this something=\"SOMETHING\"}}", &data).unwrap() = "SUB TEST //  BUS"
```

I don't know whether this is a bug, or some obscure change in the syntax that requires a change in templates, but either way it's a breaking change and I can't seem to find a way to make this work if I need to pass anything other than "this" as the model.

**Comment by sunng87** [comment]
The feature for using both hash and param context was broken during the refactoring in 4.x. This could be added back and I will be looking into this soon.

**Comment by sunng87** [comment]
Fixed in 4.3.3
